docs/cosmology/io.rst

```diff
@@ -65,12 +62,154 @@ method. The method uses this cosmology class, regardless of the class
+    
+    >>> def from_table_row(row, *, move_to_meta=False, cosmology=None):
+    ...     # get name from column
+    ...     name = row['name'] if 'name' in row.columns else None
+    ...     meta = copy.deepcopy(row.meta)
+    ...     # turn row into mapping (dict of the arguments)
+    ...     mapping = {}
+    ...     mapping["cosmology"] = meta.pop("cosmology")
+    ...     mapping.update({k: v for k, v in zip(row.colnames, row.values())})
+    ...     mapping["meta"] = meta
+    ...     # build cosmology from map
+    ...     return Cosmology.from_format(mapping,
+    ...                                  move_to_meta=move_to_meta, cosmology=cosmology)
+
+The next step is a function to perform the reverse operation: parse a
+`~astropy.cosmology.Cosmology` into a `~astropy.table.Row`. This function
+requires only the cosmology object and a ``*args`` to absorb unneeded
+information passed by `astropy.io.registry.UnifiedReadWrite` (which
+implements `astropy.cosmology.Cosmology.to_format`).
+
+.. code-block:: python
+    :emphasize-lines: 4
+
+    >>> from astropy.table import QTable
+    
+    >>> def to_table_row(cosmology, *args):
+    ...     p = cosmology.to_format("mapping")
+    ...     # create metadata from mapping
+    ...     meta = p.pop("meta")
+    ...     meta["cosmology"] = p.pop("cosmology").__name__
+    ...     # package parameters into lists for Table parsing
+    ...     params = {k: [v] for k, v in p.items()}
+    ...     return QTable(params, meta=meta)[0]  # return row
+
+Last we write a function to help with format auto-identification and then
+register everything into `astropy.io.registry`.
+
+.. code-block:: python
+    :emphasize-lines: 11, 12, 13
+
+    >>> from astropy.cosmology import Cosmology
+    >>> from astropy.io import registry as io_registry
+    >>> from astropy.table import Row
+
+    >>> def row_identify(origin, format, *args, **kwargs):
+    ...     """Identify if object uses the Table format."""
+    ...     if origin == "read":
+    ...         return isinstance(args[1], Row) and (format in (None, "row"))
+    ...     return False
+
+    >>> io_registry.register_reader("row", Cosmology, from_table_row)
+    >>> io_registry.register_writer("row", Cosmology, to_table_row)
+    >>> io_registry.register_identifier("row", Cosmology, row_identify)
+
+Now the registered functions can be used in
+:meth:`astropy.cosmology.Cosmology.from_format` and
+:meth:`astropy.cosmology.Cosmology.to_format`.
+
+.. code-block:: python
+
+    >>> from astropy.cosmology import Planck18
+    >>> row = Planck18.to_format("row")
+    >>> row
+    <Row index=0>
+      name        H0        Om0    Tcmb0    Neff    m_nu [3]    Ob0  
+             km / (Mpc s)                              eV            
+      str8     float64    float64 float64 float64   float64   float64
+    -------- ------------ ------- ------- ------- ----------- -------
+    Planck18        67.66 0.30966  2.7255   3.046 0.0 .. 0.06 0.04897
```

Comment:
Seeing this here, the distinction between `name` and `cosmology` needs to be more clear, especially in this context since people will likelly assume that `name` is the field that defines the cosmology class. I wonder if `name` should be called something more different, perhaps `id`? Or the default for `name` should be to make it all lower case so it doesn't look like a class name?

As above, I would think that the `cosmology` field should be seen here and it should be the first column.